agent/agent.go

```diff
@@ -920,6 +921,15 @@ func (a *Agent) listenAndServeDNS() error {
 			}
 		}(addr)
 	}
+	s, _ := NewDNSServer(a)
+
+	grpcDNS.NewServer(grpcDNS.Config{
+		Logger:       a.logger.Named("grpc-api.dns"),
+		DNSServeMux:  s.mux,
+		LocalAddress: grpcDNS.Local{IP: net.IPv4(127, 0, 0, 1), Port: a.config.GRPCPort},
```

Comment:
Noted. Since local address really doesn't determine much here I could also pass that port through but it seems unnecessary considering gRPC will handle the connection regardless. We just need something in local address and I don't _think_ the port will affect the underlying dns message?